geowebcache/wmts/src/main/java/org/geowebcache/service/wmts/WMTSService.java

```diff
@@ -557,6 +583,15 @@ public void handleRequest(Conveyor conv) throws OWSException, GeoWebCacheExcepti
                 ConveyorTile convTile = (ConveyorTile) conv;
                 WMTSGetFeatureInfo wmsGFI = new WMTSGetFeatureInfo(convTile);
                 wmsGFI.writeResponse(stats);
+            } else if (tile.getHint().equals(GET_TILEJSON)) {
+                getSecurityDispatcher().checkSecurity(tile);
+                ConveyorTile convTile = (ConveyorTile) conv;
+                TileLayer layer = convTile.getLayer();
+                if (layer instanceof TileJSONProvider) {
```

Comment:
Should return a 404 if someone asks for a tilejson when the url is not included in the capabilities